src/lib/viewers/doc/ExcelViewer.js

```diff
@@ -0,0 +1,179 @@
+import BaseViewer from '../BaseViewer';
+import Controls from '../../Controls';
+import { get } from '../../util';
+import './Excel.scss';
+import { VIEWER_EVENT } from '../../events';
+import { ICON_FULLSCREEN_IN, ICON_FULLSCREEN_OUT } from '../../icons/icons';
+import { DOC_STATIC_ASSETS_VERSION } from '../../constants';
+
+const JS = [
+    `third-party/doc/${DOC_STATIC_ASSETS_VERSION}/cpexcel.js`,
+    `third-party/doc/${DOC_STATIC_ASSETS_VERSION}/jszip.js`,
+    `third-party/doc/${DOC_STATIC_ASSETS_VERSION}/xlsx.full.min.js`,
+    'excel.js'
+];
+
+class ExcelViewer extends BaseViewer {
+    /**
+     * @inheritdoc
+     */
+    setup() {
+        // Call super() first to set up common layout
+        super.setup();
+
+        this.excelEl = this.containerEl.appendChild(document.createElement('div'));
+
+        // disable zooming until zooming handler is implemented
+        this.excelEl.addEventListener('touchstart', this.disableTouch);
+    }
+
+    /**
+     * [destructor]
+     *
+     * @return {void}
+     */
+    destroy() {
+        if (this.excelComponent) {
+            this.excelComponent.destroy();
+        }
+        this.excelEl.removeEventListener('touchstart', this.disableTouch);
+        super.destroy();
+    }
+
+    disableTouch(event) {
+        if (event.touches.length < 2) {
+            return;
+        }
+        event.preventDefault();
+        event.stopPropagation();
+    }
+
+    /**
+     * Disable ArrowLeft and ArrowRight for excel viewer.
+     *
+     * @param {string} key - keydown key
+     * @return {boolean} consumed or not
+     */
+    onKeydown(key) {
+        switch (key) {
```

Comment:
Don't think you need this switch statement at all for this viewer unless you plan on adding your own keyboard shortcuts. 

If you don't then just do 
```
onKeyDown(key) {
    return true;
}
```

Reply:
This way the event gets consumed w/o any actions happening